yt/loaders.py

```diff
@@ -1457,3 +1463,108 @@ def load_sample(
         loadable_path = loadable_path.joinpath(load_name, specific_file)
 
     return load(loadable_path, **kwargs)
+
+
+# --- Loader for tar-based datasets ---
+def load_archive(
+    fn: Union[str, Path],
+    path: str,
+    ratarmount_kwa: Optional[Dict] = None,
+    *args,
+    **kwargs,
+) -> Dataset:
+    r"""
+    Load archived data with yt.
+
+    This is a wrapper around :func:`~yt.loaders.load` to include mounting
+    and unmounting the archive as a read-only filesystem and load it.
+
+    Parameters
+    ----------
+
+    fn: str
+        The `filename` of the archive containing the dataset.
+
+    path: str
+        The path to the dataset in the archive.
+
+    ratarmount_kwa: dict, optional
+        Optional parameters to pass to ratarmount to mount the archive.
+
+    Notes
+    -----
+
+    - Any additional keyword argument is passed down to :func:`~yt.loaders.load`.
+    - This function requires ratarmount.
+    - This function does not work on Windows system.
+    """
+    if platform.system() not in ("Linux", "Darwin"):
+        raise RuntimeError(
+            "Reading from an archive is not supported on your operating system"
+            f" ({platform.system()})."
+        )
+
+    if isinstance(fn, Path):
+        fn = str(fn.resolve().absolute())
```

Comment:
this is the canonical way to support `os.PathLike` objects as transparently as possible (i.e. without using isinstance)
```suggestion
    fn = os.path.realpath(fn)
```